tests/testsuite/update.rs

```diff
@@ -1400,10 +1400,34 @@ fn precise_yanked() {
         .with_stderr(
             "\
 [UPDATING] `dummy-registry` index
-[ERROR] no matching package named `bar` found
-location searched: registry `crates-io`
-required by package `foo v0.0.0 ([CWD])`
+[ERROR] failed to get `bar` as a dependency of package `foo v0.0.0 ([CWD])`
+
+Caused by:
+  failed to query replaced source registry `crates-io`
+
+Caused by:
+  the `--precise <yanked-version>` flag is unstable[..]
+  See [..]
+  See [..]
+",
+        )
+        .run();
+
+    p.cargo("update --precise 0.1.1 bar")
+        .masquerade_as_nightly_cargo(&["--precise <yanked-version>"])
+        .arg("-Zunstable-options")
+        .with_stderr(
+            "\
+[UPDATING] `dummy-registry` index
+[WARNING] yanked package `bar@0.1.1` is selected by the `--precise` flag from registry `dummy-registry`
+[NOTE] it is not recommended to depend on a yanked version
+[NOTE] if possible, try other SemVer-compatbile versions
+[UPDATING] bar v0.1.0 -> v0.1.1
```

Comment:
nit: this is written from the perspective of the user knowing its yanked.  I think I would write it from the perspective that they don't know
```
[WARNING] Selected package `bar@0.1.1` was yanked by the author
[NOTE] if possible, try a compatible non-yanked version
```

Reply:
Putting `source` there was to disambiguate when needed, but maybe that's too much. And we can always improve it later on.